src/sql/azdata.proposed.d.ts

```diff
@@ -3967,7 +3967,8 @@ declare module 'azdata' {
 		export type QueryEvent =
 			| 'queryStart'
 			| 'queryStop'
-			| 'executionPlan';
+			| 'executionPlan'
+			| 'visualize';
 
 		export interface QueryEventListener {
 			onQueryEvent(type: QueryEvent, document: queryeditor.QueryDocument, args: any): void;
```

Comment:
We should add some doc comments for this event listener method, defining what type of args you can expect for a given event type. Optionally, we could make `args` be `args: ResultSetSummary | string | undefined` and explain which one it is, for each event type? At the moment you need to guess what this would be...

Reply:
Added!